api.js

```diff
@@ -1044,6 +1131,30 @@
         params.token = this.token;
       }
 
+      if (cachedRoutes[lcMethod]) {
+        var i;
+        for(i in cachedRoutes[lcMethod]) {
+          if (!cachedRoutes[lcMethod].hasOwnProperty(i)) continue;
+          var regexp = new RegExp(i);
```

Comment:
This hurts. It is better to have the regex precompiled. 
Use the .compile() to compile a regex object.